youtube_transcript_api/formatters.py

```diff
@@ -0,0 +1,174 @@
+    """Converts a `time` into a formatted transcript timecode.
+
+    :param time: a float representing time in seconds.
+    :type time: float
+    :return: a string formatted as a timecode, 'HH:MM:SS,MS'
+    :rtype str
+
+    :example:
+    >>> parse_timecode(6.93)
+    '00:00:06,930'
+    """
+    
+    time = float(time)
+    times = {
+        'hours': str(int(time) // 3600).rjust(2, '0'),
+        'mins': str(int(time) // 60).rjust(2, '0'),
+        'secs': str(int(time) % 60).rjust(2, '0'),
+        'ms': str(int(round((time - int(time))*1000, 2))).rjust(3, '0')
+    }
+    return "{hours}:{mins}:{secs},{ms}".format(**times)
+
+
+class TranscriptFormatter(object):
+    """Abstract Base TranscriptFormatter class
+
+    This class should be inherited from to create additional
+     custom transcript formatters.
+    """
+    HTML_TAG_REGEX = re.compile(r'<[^>]*>', re.IGNORECASE)
+    DELIMITER = ''
+
+    @classmethod
+    def combine(cls, transcripts):
+        """Subclass may override this class method.
+
+        Default behavior of this method will ''.join() the str() 
+         of each transcript in transcripts.
+
+        :param transcripts: a list of many transcripts
+        :type transcript_data: list[<formatted transcript>, ...]
+        :return: A string joined on the `cls.DELIMITER` to combine transcripts
+        :rtype: str
+        """
+        return cls.DELIMITER.join(
+                str(transcript) for transcript in transcripts)
+
+    @classmethod
+    def format(cls, transcript_data):
+        """Any subclass must implement this format class method.
+
+        :param transcript_data: a list of transcripts, 1 or more.
+        :type transcript_data: list[list[dict], list[dict]]
+        :return: A list where each item is an individual transcript
+         as a string.
+        :rtype: list[str]
+        """
+        raise NotImplementedError(
+            cls.__name__ + '.format'
+        )
+
+
+class JSONTranscriptFormatter(TranscriptFormatter):
+    """Formatter for outputting JSON data"""
+    DELIMITER = ','
+
+    @classmethod
+    def combine(cls, transcripts):
+        return json.dumps(transcripts)
+
+    @classmethod
+    def format(cls, transcript_data):
+        return transcript_data
+
+
+class TextTranscriptFormatter(TranscriptFormatter):
+    """Formatter for outputting a Plain Text Format
+
+    Converts the fetched transcript data into separated lines of
+     plain text separated by newline breaks (\n) with no timecodes.
+    """
+    DELIMITER = '\n\n'
+
+    @classmethod
+    def format(cls, transcript_data):
+        return '{}\n'.format('\n'.join(
+                    line['text']for line in transcript_data))
+
+
+class SRTTranscriptFormatter(TranscriptFormatter):
+    """Formatter for outputting the SRT Format
+
+    Converts the fetched transcript data into a simple .srt file format.
+    """
+    DELIMITER = '\n\n'
+
+    @classmethod
+    def format(cls, transcript_data):
```

Comment:
I don't know much about srt, so I'll just trust you on this 😄

Reply:
I am still learning about the SRT specs. So dont trust it! 🙈 

Again I used https://github.com/jdepoix/youtube-transcript-api/issues/23#issuecomment-618953435 PR as a reference for the implementation. It "appears" to be correct, but yesterday I found a bug with the actual calculated times having overlap which I dont believe is right. 

So this will be getting another pass before we merge.

Reply:
Oh, okay. Then we should probably invest some more time to check if this is actually correct. How about adding a srt file in the `test/asset` folder and asserting that the result returned by the formatter is correct?